### PR TITLE
[zh] Translate custom-resources concept

### PR DESCRIPTION
**Pull request by tengqm**
closes: #22955

### CODE REVIEW

**Review by markthink — COMMENTED** [review]
/lgtm
/approve

File: content/zh/docs/concepts/extend-kubernetes/api-extension/custom-resources.md [hunk 1]
```diff
@@ -0,0 +1,557 @@
...
+A *resource* is an endpoint in the [Kubernetes
+API](/docs/reference/using-api/api-overview/) that stores a collection of [API
+objects](/docs/concepts/overview/working-with-objects/kubernetes-objects/) of
+a certain kind; for example, the built-in *pods* resource contains a
+collection of Pod objects.
+-->
+## 定制资源
+
+*资源（Resource）* 是
+[Kubernetes API](/zh/docs/reference/using-api/api-overview/) 中的一个末端，
```
Comment (chenrui333) [inline content/zh/docs/concepts/extend-kubernetes/api-extension/custom-resources.md:40]: 可能 `端点` 更好一点？

```suggestion
[Kubernetes API](/zh/docs/reference/using-api/api-overview/) 中的一个端点，
```

File: content/zh/docs/concepts/extend-kubernetes/api-extension/custom-resources.md [hunk 2]
```diff
@@ -0,0 +1,557 @@
...
+来创建和访问其中的对象，就像他们为 *pods* 这种内置资源所做的一样。
+
+<!--
+## Custom controllers
+
+On their own, custom resources simply let you store and retrieve structured data.
+When you combine a custom resource with a *custom controller*, custom resources
+provide a true _declarative API_.
+-->
+## 定制控制器   {#custom-controllers}
```
Comment (chenrui333) [inline content/zh/docs/concepts/extend-kubernetes/api-extension/custom-resources.md:74]: 不太清楚这个 syntax
```suggestion
## 定制控制器
```

Comment (chenrui333) [reply content/zh/docs/concepts/extend-kubernetes/api-extension/custom-resources.md:74]: maybe 这个？
`[定制控制器](#custom-controllers)`

File: content/zh/docs/concepts/extend-kubernetes/api-extension/custom-resources.md [hunk 3]
```diff
@@ -0,0 +1,557 @@
...
+In a Declarative API, typically:
+
+ - Your API consists of a relatively small number of relatively small objects (resources).
+ - The objects define configuration of applications or infrastructure.
+ - The objects are updated relatively infrequently.
+ - Humans often need to read and write the objects.
+ - The main operations on the objects are CRUD-y (creating, reading, updating and deleting).
+ - Transactions across objects are not required: the API represents a desired state, not an exact state.
+-->
+### 声明式 APIs {#declarative-apis}
```
Comment (chenrui333) [inline content/zh/docs/concepts/extend-kubernetes/api-extension/custom-resources.md:152]: 和上面有类似的问题

File: content/zh/docs/concepts/extend-kubernetes/api-extension/custom-resources.md (hunk 2 above)
Comment (chenrui333) [reply content/zh/docs/concepts/extend-kubernetes/api-extension/custom-resources.md:74]: never mind, I think I understand better now.

File: content/zh/docs/concepts/extend-kubernetes/api-extension/custom-resources.md (hunk 3 above)
Comment (chenrui333) [reply content/zh/docs/concepts/extend-kubernetes/api-extension/custom-resources.md:152]: never mind, I think I understand better now.

File: content/zh/docs/concepts/extend-kubernetes/api-extension/custom-resources.md (hunk 1 above)
Comment (chenrui333) [reply content/zh/docs/concepts/extend-kubernetes/api-extension/custom-resources.md:40]: maybe just this one

Comment (tengqm) [reply content/zh/docs/concepts/extend-kubernetes/api-extension/custom-resources.md:40]: OK

**Comment by tengqm** [comment]
@chenrui333 Feedbacks addressed.

**Review by tanjunchen — APPROVED** [review]
/lgtm

**Pull request merged**